shell/platform/android/io/flutter/plugin/editing/TextEditingDelta.java

```diff
@@ -0,0 +1,106 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugin.editing;
+
+import io.flutter.Log;
+
+public class TextEditingDelta {
+  private CharSequence oldText;
+  private CharSequence deltaText;
+  private CharSequence deltaType;
+  private int deltaStart;
+  private int deltaEnd;
+
+  public TextEditingDelta() {
+    oldText = "";
+    deltaText = "";
+    deltaType = "EQUALITY";
+    deltaStart = -1;
+    deltaEnd = -1;
+  }
+
+  public CharSequence getOldText() {
+    return oldText;
+  }
+
+  public CharSequence getDeltaText() {
+    return deltaText;
+  }
+
+  public CharSequence getDeltaType() {
+    return deltaType;
+  }
+
+  public int getDeltaStart() {
+    return deltaStart;
+  }
+
+  public int getDeltaEnd() {
+    return deltaEnd;
+  }
+
+  private void setDeltas(
+      CharSequence oldTxt, CharSequence newTxt, CharSequence type, int newStart, int newExtent) {
+    oldText = oldTxt;
+    deltaText = newTxt;
+    deltaStart = newStart;
+    deltaEnd = newExtent;
+    deltaType = type;
+  }
+
+  public void reasonDeltas(
```

Comment:
When you get to writing tests, it will be nice to be able to test all of this logic directly without needing to spin up a TextInputChannel and everything.